os/fs/driver/mtd/smart.c

```diff
@@ -4395,31 +4395,28 @@ static int smart_readsector(FAR struct smart_struct_s *dev, unsigned long arg)
 #endif
 	if (physsector == 0xFFFF) {
 		fdbg("Logical sector %d not allocated\n", req->logsector);
-		return -EINVAL;
+		ret = -EINVAL;
+		goto errout;
 	}
-#ifdef CONFIG_MTD_SMART_ENABLE_CRC
 
-	/* When CRC is enabled, we read the entire sector into RAM so we can validate the CRC.
-	 * However, if Journaling is enabled, written data is read back and verified with CRC,
-	 * so we can skip this step.
-	 */
+/* If journaling is disabled, data is not read back and verified at the time of writing itself. Hence we need to read the entire sector into RAM to validate CRC. */
+#if defined(CONFIG_MTD_SMART_ENABLE_CRC) && !defined(CONFIG_MTD_SMART_JOURNALING)
 
 	ret = MTD_BREAD(dev->mtd, physsector * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)dev->rwbuffer);
 	if (ret != dev->mtdBlksPerSector) {
 		fdbg("Error reading phys sector %d\n", physsector);
-		return -EIO;
+		ret = -EIO;
+		goto errout;
```

Comment:
There is no cleanup code at  `errout`. So i think we can directly return from here.